contracts/exchangeIssuance/ExchangeIssuancePerp.sol

```diff
@@ -0,0 +1,298 @@
+/*
+    Copyright 2022 Index Cooperative
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+pragma experimental ABIEncoderV2;
+
+import { IQuoter } from "../interfaces/IQuoter.sol";
+import { IUniswapV3SwapCallback } from "../interfaces/IUniswapV3SwapCallback.sol";
+import { ISwapRouter} from "../interfaces/external/ISwapRouter.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { Context } from "@openzeppelin/contracts/GSN/Context.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { SignedSafeMath } from "@openzeppelin/contracts/math/SignedSafeMath.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { TransferHelper } from "../lib/TransferHelper.sol";
+import { ISetToken } from "../interfaces/ISetToken.sol";
+import { IDebtIssuanceModule } from "../interfaces/IDebtIssuanceModule.sol";
+import { ISlippageIssuanceModule } from "../interfaces/ISlippageIssuanceModule.sol";
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import { Withdrawable } from "external/contracts/aaveV2/utils/Withdrawable.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/SafeCast.sol";
+import { PreciseUnitMath } from "../lib/PreciseUnitMath.sol";
+
+
+// Flash issue basis trading products using SlippageIssuanceModule
```

Comment:
It'd be nice to have the styling of comments etc. consistent with our other contracts.
Also please add full documentation comments to all external methods.

See the leveraged EI module for reference:
https://github.com/IndexCoop/index-coop-smart-contracts/blob/master/contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol

Reply:
https://github.com/IndexCoop/index-coop-smart-contracts/pull/111/commits/cc4813459ab66c04ea9a708855e46a7f4eda0ec4